src/relativedeltafield/__init__.py

```diff
@@ -147,3 +169,20 @@ def from_db_value(self, value, expression, connection, context=None):
 	def value_to_string(self, obj):
 		val = self.value_from_object(obj)
 		return '' if val is None else format_relativedelta(val)
+
+	def formfield(self, *args, **kwargs):
+		kwargs.setdefault('choices_form_class', relativedeltafield.forms.RelativeDeltaChoiceField)
+		return super().formfield(*args, **kwargs)
+
+
+if has_migration_serializer:
+	# Django 2.2 and up
+	class RelativeDeltaSerializer(BaseSerializer):
+		def serialize(self):
+			return repr(self.value), {'from dateutil.relativedelta import relativedelta'}
+
+	MigrationWriter.register_serializer(relativedelta, RelativeDeltaSerializer)
+
+elif has_migration_deconstructible:
+	# Django 2.1 and lower
+	deconstructible(relativedelta)
```

Comment:
Why is this migration stuff needed, exactly?

I've used django-relativedelta in a few projects and it never required any special code.

Reply:
`makemigrations` does not know how to serialise `relativedelta` if the model fields' arguments contains a `relativedelta` object, and this tells it how. `choices` are defined by a "list of (relativedelta, str) pair", so we need this when `choices` is defined. Without this, you'd get an error when doing `makemigrations`:

    ValueError: Cannot serialize: relativedelta(days=+1)
    There are some values Django cannot serialize into migration files.
    For more, see https://docs.djangoproject.com/en/3.1/topics/migrations/#migration-serializing

Without this serialisation stuffs, `default` wouldn't work either `RelativeDeltaField(default=relativedelta(days=1))`.